modules/tensor_mechanics/src/materials/ADLAROMANCEStressUpdateBase.C

```diff
@@ -692,10 +692,13 @@ ADLAROMANCEStressUpdateBase::convertOutput(const std::vector<Real> & old_input_v
     return 0.0;
 
   ADReal expout = std::exp(rom_output);
-  if (expout > 1.0e-10)
-    expout -= 1.000000E-10;
+  mooseAssert(expout > 0.0, "ROM calculated strain increment is zero or negative");
```

Comment:
How is this assertion ever hit?! The return value of `std::exp` should always be > 0... unless `rom_output` is `NaN` or (maybe) `-inf`.

Reply:
It can certainly never be negative

Reply:
> We divide by expout below without further checks. I suggest adding an assertion to make sure expout > 0

Did you want the assert from the Oct 14 review above the exponential call? I'm happy to move it and I'm just not sure where you'd like to see the assert

Reply:
Ah, touche :-). How about changing the assertion text to "is not strictly positive"?

Reply:
done